plugins/newsql/newsql.c

```diff
@@ -1989,9 +1995,14 @@ static CDB2QUERY *read_newsql_query(struct dbenv *dbenv,
         Pthread_mutex_unlock(&clnt->wait_mutex);
     }
 
-    if (!query) return NULL;
+    if (!query) {
+        logmsg(LOGMSG_ERROR, "query missing\n");
+        return NULL;
+    }
+
     if (errno != 0) {
         cdb2__query__free_unpacked(query, &pb_alloc);
+        logmsg(LOGMSG_ERROR, "errno: %d\n", errno);
```

Comment:
A bit more context would make it clearer.

Reply:
Done